gravitee-apim-gateway/gravitee-apim-gateway-http/src/main/java/io/gravitee/gateway/reactive/http/vertx/VertxHttpServerRequest.java

```diff
@@ -196,8 +200,11 @@ public String host() {
     }
 
     @Override
-    public Maybe<Buffer> body() {
-        return chunks.reduce(Buffer::appendBuffer);
+    public synchronized Maybe<Buffer> body() {
```

Comment:
There is no real concurrency here so it is not so expensive. We shouldn't have concurrency because policies, processors, ... are executed sequentially but it still possible so I prefer synchronize to be sure.

Reply:
Because the we are in a reactive chain were contributors are policies and they can do things outside of the current thread so technically it is possible to replace the body at the same time from 2 different threads (even if it stil highly improbable).

Reply:
I've removed the synchronize for now. As we are not in a highly competitive situation, it is not a pre-requisite for now.  We will check later if it is something to introduce or not.